tools/debug_tools/crash_log_parser/crash_log_parser.py

```diff
@@ -129,149 +111,110 @@ def parseCPUID(cpuid):
+    
+    print("\n\nCrash Info:")        
     print("\tCrash location = %s [%s] (based on PC value)"%(pc_name.strip(),str(pc_val)))
     print("\tCaller location = %s [%s] (based on LR value)"%(lr_name.strip(),str(lr_val)))        
     print("\tStack Pointer at the time of crash = [%s]"%(str(sp_val)))
     
-    print("\tTarget/Fault Info:")
+    print("\tTarget and Fault Info:")
     parseCPUID(cpuid_val)
     parseHFSR(hfsr_val)
     parseMMFSR(mmfsr_val,mmfar_val)
     parseBFSR(bfsr_val,bfar_val)
     parseUFSR(ufsr_val)
         
-    print("\nDone parsing...")    
-        
                             
 if __name__ == '__main__':
     import argparse
     
-    parser = argparse.ArgumentParser(
-        description='Analyse mbed-os crash log')
-
+    parser = argparse.ArgumentParser(description='Analyse mbed-os crash log. This tool requires arm-gcc binary utilities to be available in current path as it uses \'nm\' command')
     # specify arguments
-    parser.add_argument('-i','--input', metavar='<path to input file with crash log output>', type=str,
+    parser.add_argument('-i','--input', metavar='<path to input file with crash log output>', type=argparse.FileType('rb', 0), required=True,
                        help='path to input file')      
-    parser.add_argument('-e','--elfpath', metavar='<module or elf file path>', type=str,
+    parser.add_argument('-e','--elfpath', metavar='<module or elf file path>', type=argparse.FileType('rb', 0), required=True,
                        help='path to elf file')             
-    parser.add_argument('-m','--mappath', metavar='<map file path>', type=str,
+    parser.add_argument('-m','--mappath', metavar='<map file path>', type=argparse.FileType('rb', 0), required=True,
                        help='path to map file')                                    
 
     # get and validate arguments
     args = parser.parse_args()
     
-    p = args.elfpath
-    m = args.mappath
-    i = args.input
-    if(p == None or m == None or i == None):
-        print("Invalid arguments, exiting")
-        parser.print_usage()
-        sys.exit(1)
-        
-    if not path.exists(p):
-        print("Elf path %s does not exist"%(str(p)))
-        parser.print_usage()
-        sys.exit(1)
-        
-    if not path.exists(m):
-        print("Map path %s does not exist"%(str(m)))
-        parser.print_usage()
-        sys.exit(1)    
-        
-    if not path.exists(i):
-        print("Input crash log path %s does not exist"%(str(i)))
-        parser.print_usage()
-        sys.exit(1) 
+    elf_file = args.elfpath
+    map_file = args.mappath
+    input_crash_log = args.input
 
     print("Inputs:")        
-    print("\tCrash Log: %s"%(i))        
-    print("\tElf Path: %s"%(p))        
-    print("\tMap Path: %s"%(m))        
+    print("\tCrash Log: %s"%(input_crash_log.name))        
+    print("\tElf Path: %s"%(elf_file.name))        
+    print("\tMap Path: %s"%(map_file.name))        
     
```

Comment:
@theotherjimmy  The above prints are not for debugging but just to make sure the user sees what input files they are passing in as people operate with multiple builds/toolchains and their paths could be very similar.